packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -712,8 +712,12 @@ function createAsyncData<
       asyncData._init = false
       // TODO: disable in v4 in favour of custom caching strategies
       if (purgeCachedData && !hasCustomGetCachedData) {
-        clearNuxtDataByKey(nuxtApp, key)
-        asyncData.execute = () => Promise.resolve()
+        nextTick(() => {
+          if (!asyncData._init) {
```

Comment:
Hi @danielroe! I guess it was meant to be:
`if (!nuxtApp._asyncData[key]?._init) {`

I'm not sure, but I don't see any place where _init could be switched from `false` to `true`. And it seems like there can be a bug if an asyncData is going to be cleared and a new asyncData with the same key is created at this moment.